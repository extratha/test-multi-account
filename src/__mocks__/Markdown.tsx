const MockMarkdown = (props: any) => {
  return <div {...props} />;
};

jest.mock("react-markdown", () => MockMarkdown);
