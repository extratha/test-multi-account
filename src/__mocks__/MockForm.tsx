import { yupResolver } from "@hookform/resolvers/yup";
import { ReactNode, useEffect } from "react";
import { FormProvider, UseFormReturn, useForm } from "react-hook-form";
import { ObjectSchema } from "yup";

export interface MockFormProps<T = any> {
  initValues: T | undefined;
  resolver?: ObjectSchema<any>;
  children?: ReactNode;
  onInitForm?: (methods: UseFormReturn) => void;
  onSubmit: () => void;
}

export interface MockHandleValueProps {
  name: string;
}

const MockForm = (props: MockFormProps) => {
  const { initValues, resolver, children, onInitForm, onSubmit } = props;
  const methods = useForm({
    reValidateMode: "onSubmit",
    defaultValues: initValues,
    resolver: resolver ? yupResolver(resolver) : undefined,
  });

  useEffect(() => {
    if (onInitForm) onInitForm(methods);
  }, []);

  return (
    <FormProvider {...methods}>
      <form noValidate onSubmit={methods.handleSubmit(onSubmit)}>
        {children}
        <button type="submit" data-testid="submit-button">
          Submit
        </button>
      </form>
    </FormProvider>
  );
};

export default MockForm;
