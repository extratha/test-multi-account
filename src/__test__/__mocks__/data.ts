export const data = {
  test: 'test'
}