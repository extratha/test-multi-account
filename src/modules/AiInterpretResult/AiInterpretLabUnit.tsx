import { Stack, Typography } from "@mui/material";
import { useTranslations } from "next-intl";

import { InputData } from "@/types/aiInterpret";

export interface AiInterpretLabUnitProps {
  name: string;
  groupName: string;
  inputData: InputData;
}

const AiInterpretLabUnit = (props: AiInterpretLabUnitProps) => {
  const { name, groupName, inputData } = props;
  const tAi = useTranslations("AiInterpret");

  const getValue = () => {
    const unit = inputData.unit && `(${inputData.unit})`;
    return `${inputData.value}${unit}`;
  };

  return (
    <Stack direction="row" justifyContent="space-between">
      <Stack width="100%" spacing="8px">
        <Typography
          variant="titleMedium"
          fontSize="16px"
          fontWeight={700}
          data-testid={`${name}-${groupName}-${inputData.key}-label`}
        >
          {tAi(`aiInterpretResult.lab.group.${groupName}.${inputData.key}.label`)}
        </Typography>
      </Stack>
      <Stack width="25%">
        <Typography
          variant="titleMedium"
          fontWeight={700}
          textAlign="end"
          data-testid={`${name}-${groupName}-${inputData.key}-value`}
        >
          {getValue()}
        </Typography>
        <Typography variant="bodyMedium" textAlign="end" data-testid={`${name}-${groupName}-${inputData.key}-unit`}>
          {inputData.unit && `(${inputData.range[0].value} ${inputData.unit})`}
        </Typography>
      </Stack>
    </Stack>
  );
};

export default AiInterpretLabUnit;
