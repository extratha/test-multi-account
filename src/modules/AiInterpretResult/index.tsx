"use client";

import { Button, CircularProgress, Divider, Stack, Typography } from "@mui/material";
import { styled } from "@mui/material/styles";
import { useTranslations } from "next-intl";
import { useRouter, useSearchParams } from "next/navigation";
import { useMemo } from "react";
import ReactMarkdown from "react-markdown";

import { IconArrowLeft, IconCopy, IconPen, IconSparkle } from "@/assets";
import Tag from "@/components/Tag";
import { CUSTOM_COLORS, NEUTRAL, linearGradient } from "@/config/config-mui/theme/colors";
import { GENDER, GENERAL_CHECK_UP, GROUP_NAME } from "@/constant/constant";
import { webPaths } from "@/constant/webPaths";
import { useGetLabExampleId } from "@/hooks/useApi";
import { GroupName, InputData } from "@/types/aiInterpret";
import AiInterpretLabResult from "./AiInterpretLebResult";

const ContentContainer = styled(Stack)({
  height: "100%",
  width: "100%",
  padding: "24px",
});

const ContentContainerWrapper = styled(Stack)(({ theme }) => ({
  maxWidth: "1080px",
  margin: "auto",
  padding: "24px 30px",
  borderRadius: "16px",
  overflowY: "auto",
  backgroundColor: theme.palette.background.paper,
  "& .MuiButton-root": {
    width: "fit-content;",
    height: "40px",
    padding: "10px 16px",
    alignItems: "center",
    borderRadius: "8px",
    border: "1px solid #CFD8DC",
    fontSize: "12px",
    fontWeight: 700,
    color: CUSTOM_COLORS.textHighEmp,
  },
  "& > .MuiButton-startIcon": {
    marginRight: "8px",
  },
}));

const DividerLine = styled(Divider)({
  margin: "8px 0px",
});

const InformationBackground = styled(Stack)(({ theme }) => ({
  justifyContent: "center",
  alignItems: "center",
  padding: "2px",
  borderRadius: "24px",
  background: theme.palette.gradients?.main,
}));

const InformationBox = styled(Stack)(({ theme }) => ({
  width: "100%",
  justifyContent: "start",
  borderRadius: "22px",
  background: theme.palette.common.white,
}));

const Example = styled(Typography)(({ theme }) => ({
  fontWeight: 600,
  color: theme.palette.primary.lighter,
}));

const ModelVersion = styled(Stack)(({ theme }) => ({
  color: theme.palette.primary.lighter,
}));

const Version = styled(Typography)({
  fontWeight: 600,
});

const Name = styled(Typography)({
  fontWeight: 700,
  color: CUSTOM_COLORS.textHighEmp,
});

const InterpretTile = styled(Stack)(({ theme }) => ({
  padding: "8px 16px",
  alignItems: "center",
  alignSelf: "center",
  borderRadius: "32px",
  background: theme.palette.gradients?.main,
}));

const DividerDashed = styled("div")({
  padding: "1px 0px",
  margin: "0px 0px -18px 0px",
  backgroundImage: linearGradient.dashed,
  backgroundPosition: "top",
  backgroundSize: "10px 1px",
  backgroundRepeat: "repeat-x",
});

const Title = styled(Typography)(({ theme }) => ({
  fontWeight: 700,
  color: theme.palette.common.white,
}));

const TitleResult = styled(Typography)(({ theme }) => ({
  fontWeight: 600,
  color: theme.palette.primary.lighter,
}));

const Markdown = styled(ReactMarkdown)({
  fontSize: "16px",
  whiteSpace: "normal",
  color: CUSTOM_COLORS.textHighEmp,
});

const General = styled(Stack)({
  color: CUSTOM_COLORS.textHighEmp,
  border: `1px solid ${NEUTRAL[99]}`,
  borderRadius: "8px",
  overflow: "hidden",
});

const GeneralHeader = styled(Stack)({
  padding: "24px 24px 12px 24px",
  backgroundColor: NEUTRAL[97],
});

const GeneralTitle = styled(Typography)({
  fontSize: "18px",
  fontWeight: 700,
});

const GeneralInformation = styled(Stack)({
  padding: "12px 24px 24px",
});

const GeneralInformationUnit = styled(Typography)({
  width: "60px",
  color: CUSTOM_COLORS.lightSteelgray,
});

const AiInterpretResult = () => {
  const searchParams = useSearchParams();
  const interpretId = searchParams.get("id") || "";
  const tAi = useTranslations("AiInterpret");
  const router = useRouter();

  const { data, isLoading } = useGetLabExampleId(interpretId);
  const interpretData = data?.data;

  const aiResultData = interpretData?.aiResult.data || [];
  const inputDataResultData = interpretData?.inputData || [];

  const gender: Record<string, string> = {
    [GENDER.MALE]: tAi("aiInterpretResult.gender.male"),
    [GENDER.FEMALE]: tAi("aiInterpretResult.gender.female"),
  };

  const informationGender: Record<string, string> = {
    [GENDER.MALE]: tAi("aiInterpretResult.general.information.gender.male"),
    [GENDER.FEMALE]: tAi("aiInterpretResult.general.information.gender.female"),
  };

  const { generalData, labData } = useMemo(() => {
    const generalData: GroupName[] = [];
    const labData: GroupName[] = [];

    inputDataResultData.forEach((item) => {
      if (item.groupName === GROUP_NAME.GENERAL_CHECK_UP) {
        generalData.push(item);
      } else {
        labData.push(item);
      }
    });

    return { generalData, labData };
  }, [inputDataResultData.length]);

  const handleClickEdit = () => {
    router.push(`${webPaths.aiInterpret.tryInputData}?id=${interpretId}`);
  };

  const handleClickBack = () => {
    router.push(webPaths.aiInterpret.tryExampleData);
  };

  const handleClickCopy = (text: string) => {
    navigator.clipboard.writeText(text);
  };

  const getInformationValue = (item: InputData) => {
    switch (item.key) {
      case GENERAL_CHECK_UP.GENDER:
        return informationGender[item.value];
      default:
        return item.value;
    }
  };

  const getInformationUnit = (item: InputData) => {
    switch (item.key) {
      case GENERAL_CHECK_UP.AGE:
        return tAi("aiInterpretResult.general.information.age.unit");
      default:
        return item.unit;
    }
  };

  return (
    <ContentContainer>
      {isLoading ? (
        <Stack width="100%" height="100%" data-testid="ai-interpret-loading">
          <CircularProgress style={{ margin: "auto" }} data-testid="ai-interpret-circular-progress" />
        </Stack>
      ) : (
        <ContentContainerWrapper>
          <Stack direction="row" justifyContent="space-between">
            <Button startIcon={<IconArrowLeft />} onClick={handleClickBack} data-testid="ai-interpret-button-back">
              {tAi("aiInterpretResult.button.back")}
            </Button>
            <Button startIcon={<IconPen />} onClick={handleClickEdit} data-testid="ai-interpret-button-edit">
              {tAi("aiInterpretResult.button.edit")}
            </Button>
          </Stack>
          <DividerLine />
          <Stack spacing="16px" padding="16px 0px">
            <InformationBackground>
              <InformationBox>
                <Stack padding="24px" spacing="8px">
                  <Stack direction="row" justifyContent="space-between">
                    <Example variant="labelMedium" data-testid="ai-interpret-example-rank">
                      {tAi("aiInterpretResult.example", { num: interpretData?.ranking })}
                    </Example>
                    <ModelVersion direction="row" spacing="4px">
                      <Typography variant="bodySmall">{tAi("aiInterpretResult.modalVersion")}</Typography>
                      <Version variant="labelMedium" data-testid="ai-interpret-version">
                        {interpretData?.aiModelVersion}
                      </Version>
                    </ModelVersion>
                  </Stack>
                  <Name variant="titleLargeSemiBold" data-testid="ai-interpret-example-name">
                    {interpretData?.caseName}
                  </Name>
                  <Stack direction="row" spacing="8px">
                    <Tag name="ai-interpret-gender" text={gender[interpretData?.gender || ""]} />
                    <Tag name="ai-interpret-age" text={tAi("aiInterpretResult.age", { age: interpretData?.age })} />
                  </Stack>
                </Stack>
                {aiResultData.length > 0 && (
                  <>
                    <Stack padding="16px 0px 0px 0px">
                      <DividerDashed />
                      <InterpretTile direction="row" spacing="6px">
                        <IconSparkle />
                        <Title variant="labelMedium" data-testid="ai-interpret-title">
                          {tAi("aiInterpretResult.title")}
                        </Title>
                      </InterpretTile>
                    </Stack>
                    <Stack padding="24px" spacing="24px" divider={<Divider flexItem />}>
                      {aiResultData.map((option, index) => (
                        <Stack key={`ai-interpret-result-${index}`} spacing="16px">
                          <TitleResult variant="labelExtraLargeSemiBold" data-testid={`ai-interpret-title-${index}`}>
                            {index + 1}. {option.title}
                          </TitleResult>
                          <Markdown data-testid={`ai-interpret-description-${index}`}>{option.description}</Markdown>
                          <Button
                            startIcon={<IconCopy />}
                            onClick={() => {
                              handleClickCopy(option.description);
                            }}
                            data-testid={`ai-interpret-button-copy-${index}`}
                          >
                            {tAi("aiInterpretResult.button.copy")}
                          </Button>
                        </Stack>
                      ))}
                    </Stack>
                  </>
                )}
              </InformationBox>
            </InformationBackground>
            {inputDataResultData.length > 0 && (
              <General>
                <GeneralHeader>
                  <GeneralTitle variant="titleMediumBold">{tAi("aiInterpretResult.general.title")}</GeneralTitle>
                </GeneralHeader>
                {generalData[0].data.map((item, index) => (
                  <GeneralInformation key={`general-${index}`} direction="row" justifyContent="space-between">
                    <Typography
                      variant="titleMediumBold"
                      fontWeight={700}
                      data-testid={`ai-interpret-general-information-title-${item.key}`}
                    >
                      {tAi(`aiInterpretResult.general.information.${item.key}.title`)}
                    </Typography>

                    <Stack direction="row" spacing="36px">
                      <Typography
                        variant="bodyMedium"
                        data-testid={`ai-interpret-general-information-${item.key}-value`}
                      >
                        {getInformationValue(item)}
                      </Typography>
                      <GeneralInformationUnit
                        variant="bodyMedium"
                        data-testid={`ai-interpret-general-information-${item.key}-unit`}
                      >
                        {getInformationUnit(item)}
                      </GeneralInformationUnit>
                    </Stack>
                  </GeneralInformation>
                ))}
              </General>
            )}
            {labData.map((item, index) => (
              <AiInterpretLabResult key={`ai-interpret-lab-${index}`} name="ai-interpret-lab" group={item} />
            ))}
          </Stack>
        </ContentContainerWrapper>
      )}
    </ContentContainer>
  );
};

export default AiInterpretResult;
