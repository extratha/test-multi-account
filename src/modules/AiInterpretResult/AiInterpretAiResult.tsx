import { Button, Stack, Typography } from "@mui/material";
import { styled } from "@mui/material/styles";
import { useMemo, useState } from "react";
import Markdown from "react-markdown";

import { IconCopy, IconGlobe } from "@/assets";
import { LANGUAGE } from "@/constant";
import useTranslation from "@/locales/useLocale";
import { AiData } from "@/types/model.api";

export interface AiInterpretAiResultProps {
  name: string;
  title: string;
  data: AiData;
}

const TitleResult = styled(Typography)(({ theme }) => ({
  fontWeight: 600,
  color: theme.palette.primary.main,
}));

const AiInterpretAiResult = ({ name, title, data }: AiInterpretAiResultProps) => {
  const { translation } = useTranslation();
  const [language, setLanguage] = useState(LANGUAGE.TH);

  const description = useMemo(() => {
    const text = data.descriptions.find((item) => item.language === language);
    return text?.description || "";
  }, [language]);

  const handleClickTranslate = () => {
    setLanguage(language === LANGUAGE.TH ? LANGUAGE.EN : LANGUAGE.TH);
  };

  const handleClickCopy = (text: string) => {
    navigator.clipboard.writeText(`${data.title}\n${text}`);
  };

  return (
    <Stack spacing="16px">
      <TitleResult data-testid={`${name}-title`}>{title}</TitleResult>
      <Markdown data-testid={`${name}-description`}>{description}</Markdown>
      <Stack direction="row" spacing="16px">
        <Button
          startIcon={<IconCopy />}
          onClick={() => handleClickCopy(description)}
          data-testid={`${name}-button-copy`}
        >
          {translation("AiInterpret.aiInterpretResult.button.copy")}
        </Button>
        {data.descriptions.length > 1 && (
          <Button
            startIcon={<IconGlobe />}
            onClick={handleClickTranslate}
            data-testid={`${name}-change-translate-button`}
          >
            {translation("AiInterpret.aiInterpretResult.button.multiple")}
          </Button>
        )}
      </Stack>
    </Stack>
  );
};

export default AiInterpretAiResult;
