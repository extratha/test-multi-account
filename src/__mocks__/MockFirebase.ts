import { mockInputFieldConfig } from "./data";

jest.mock("firebase/app", () => ({
  initializeApp: jest.fn(),
}));

jest.mock("firebase/remote-config", () => ({
  getRemoteConfig: jest.fn().mockReturnValue({ settings: {} }),
  fetchAndActivate: jest.fn().mockResolvedValue(""),
  getValue: jest.fn().mockReturnValue({
    asString: () => mockInputFieldConfig,
  }),
}));
