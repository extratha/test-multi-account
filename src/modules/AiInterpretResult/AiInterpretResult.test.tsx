import MockAdapter from "axios-mock-adapter";

import { mockAiInterpretResult } from "@/__tests__/__mocks__/data";
import {
  API,
  SpyUseRouter,
  flushPromise,
  render,
  screen,
  spyUseParams,
  spyUseRouter,
  userEvent,
} from "@/__tests__/testUtils";
import axiosInstance from "@/utils/axios";
import AiInterpretResult, { ConfigurationInterpretParams } from ".";

describe("AiInterpretResult", () => {
  let params: ConfigurationInterpretParams;
  let mockApiAdapter: MockAdapter;
  let spyRouter: SpyUseRouter;

  beforeEach(() => {
    params = {
      interpretId: "interpretId",
    };

    spyUseParams().mockReturnValue(params);
    spyRouter = spyUseRouter();

    mockApiAdapter = new MockAdapter(axiosInstance);
    mockApiAdapter.onGet(API.AI_INTERPRET_URL).reply(200, mockAiInterpretResult);

    Object.assign(navigator, {
      clipboard: {
        writeText: jest.fn(),
      },
    });
  });

  const renderAiInterpretResult = async () => {
    const view = render(<AiInterpretResult />);
    await flushPromise();
    return view;
  };

  it("should render correctly", async () => {
    const { asFragment } = await renderAiInterpretResult();
    expect(asFragment()).toMatchSnapshot();
  });

  it("should navigate to example data when click back button", async () => {
    await renderAiInterpretResult();

    await userEvent.click(screen.getByTestId("ai-interpret-button-back"));

    expect(spyRouter.push).toHaveBeenCalled();
  });

  it("should copy to example data when click copy button", async () => {
    await renderAiInterpretResult();

    await userEvent.click(screen.getByTestId("ai-interpret-button-copy-0"));

    expect(navigator.clipboard.writeText).toHaveBeenCalledWith("description-ai-result");
  });
});
