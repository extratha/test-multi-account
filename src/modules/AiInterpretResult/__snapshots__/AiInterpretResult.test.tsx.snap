// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AiInterpretResult should render correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  width: 100%;
  padding: 24px;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  padding: 24px 30px;
  border-radius: 16px;
  overflow-y: auto;
  background-color: #fff;
}

.emotion-1 .MuiButton-root {
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  height: 40px;
  padding: 10px 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 8px;
  border: 1px solid #CFD8DC;
  font-size: 12px;
  font-weight: 700;
  color: #212121;
}

.emotion-1>.MuiButton-startIcon {
  margin-right: 8px;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
}

.emotion-3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: NotoSansThai-Regular,sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 8px;
  border-radius: 8px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #1976d2;
}

.emotion-3::-moz-focus-inner {
  border-style: none;
}

.emotion-3.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-3 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-3:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(25, 118, 210, 0.04);
}

@media (hover: none) {
  .emotion-3:hover {
    background-color: transparent;
  }
}

.emotion-3.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
}

.emotion-4 {
  display: inherit;
  margin-right: 8px;
  margin-left: -4px;
}

.emotion-4>*:nth-of-type(1) {
  font-size: 20px;
}

.emotion-5 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-9 {
  margin: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  border-width: 0;
  border-style: solid;
  border-color: rgba(0, 0, 0, 0.12);
  border-bottom-width: thin;
  margin: 8px 0px;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 16px 0px;
}

.emotion-10>:not(style):not(style) {
  margin: 0;
}

.emotion-10>:not(style)~:not(style) {
  margin-top: 16px;
}

.emotion-11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 2px;
  border-radius: 24px;
}

.emotion-12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: start;
  justify-content: start;
  border-radius: 22px;
  background: #fff;
}

.emotion-13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 24px;
}

.emotion-13>:not(style):not(style) {
  margin: 0;
}

.emotion-13>:not(style)~:not(style) {
  margin-top: 8px;
}

.emotion-15 {
  margin: 0;
  font-weight: 500;
  line-height: 16px;
  letter-spacing: 0.5px;
  font-size: 12px;
  font-family: NotoSansThai-Medium,sans-serif;
  font-weight: 600;
}

.emotion-16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-16>:not(style):not(style) {
  margin: 0;
}

.emotion-16>:not(style)~:not(style) {
  margin-left: 4px;
}

.emotion-17 {
  margin: 0;
  font-weight: 400;
  line-height: 16px;
  letter-spacing: 0.4px;
  font-size: 12px;
  font-family: NotoSansThai-Regular,sans-serif;
}

.emotion-19 {
  margin: 0;
  font-weight: 500;
  line-height: 28px;
  letter-spacing: 0;
  font-size: 20px;
  font-family: NotoSansThai-SemiBold,sans-serif;
  font-weight: 700;
  color: #212121;
}

.emotion-20 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-20>:not(style):not(style) {
  margin: 0;
}

.emotion-20>:not(style)~:not(style) {
  margin-left: 8px;
}

.emotion-21 {
  padding: 4px 8px;
  background: #ECEFF1;
  border-radius: 4px;
}

.emotion-22 {
  margin: 0;
  font-weight: 500;
  line-height: 20px;
  letter-spacing: 0.1px;
  font-size: 14px;
  font-family: NotoSansThai-SemiBold,sans-serif;
}

.emotion-25 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 16px 0px 0px 0px;
}

.emotion-26 {
  padding: 1px 0px;
  margin: 0px 0px -18px 0px;
  background-image: linear-gradient(to right, #CFD8DC 50%, #ffffff 0%);
  -webkit-background-position: top;
  background-position: top;
  -webkit-background-size: 10px 1px;
  background-size: 10px 1px;
  background-repeat: repeat-x;
}

.emotion-27 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 8px 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  border-radius: 32px;
}

.emotion-27>:not(style):not(style) {
  margin: 0;
}

.emotion-27>:not(style)~:not(style) {
  margin-left: 6px;
}

.emotion-28 {
  margin: 0;
  font-weight: 500;
  line-height: 16px;
  letter-spacing: 0.5px;
  font-size: 12px;
  font-family: NotoSansThai-Medium,sans-serif;
  font-weight: 700;
  color: #fff;
}

.emotion-29 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 24px;
}

.emotion-29>:not(style):not(style) {
  margin: 0;
}

.emotion-29>:not(style)~:not(style) {
  margin-top: 24px;
}

.emotion-30 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-30>:not(style):not(style) {
  margin: 0;
}

.emotion-30>:not(style)~:not(style) {
  margin-top: 16px;
}

.emotion-31 {
  margin: 0;
  font-weight: 500;
  line-height: 24px;
  letter-spacing: 0.1px;
  font-size: 16px;
  font-family: NotoSansThai-SemiBold,sans-serif;
  font-weight: 600;
}

.emotion-32 {
  margin: 0;
  font-family: NotoSansThai-Regular,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  color: #212121;
}

<div
    class="MuiStack-root emotion-0"
  >
    <div
      class="MuiStack-root emotion-1"
    >
      <div
        class="MuiStack-root emotion-2"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium emotion-3"
          data-testid="ai-interpret-button-back"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-startIcon MuiButton-iconSizeMedium emotion-4"
          >
            <svg />
          </span>
          Back to menu
          <span
            class="MuiTouchRipple-root emotion-5"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium emotion-3"
          data-testid="ai-interpret-button-edit"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-startIcon MuiButton-iconSizeMedium emotion-4"
          >
            <svg />
          </span>
          Edit
          <span
            class="MuiTouchRipple-root emotion-5"
          />
        </button>
      </div>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth emotion-9"
      />
      <div
        class="MuiStack-root emotion-10"
      >
        <div
          class="MuiStack-root emotion-11"
        >
          <div
            class="MuiStack-root emotion-12"
          >
            <div
              class="MuiStack-root emotion-13"
            >
              <div
                class="MuiStack-root emotion-2"
              >
                <span
                  class="MuiTypography-root MuiTypography-labelMedium emotion-15"
                  data-testid="ai-interpret-example-rank"
                >
                  Example data 1
                </span>
                <div
                  class="MuiStack-root emotion-16"
                >
                  <span
                    class="MuiTypography-root MuiTypography-bodySmall emotion-17"
                  >
                    Model Version
                  </span>
                  <span
                    class="MuiTypography-root MuiTypography-labelMedium emotion-15"
                    data-testid="ai-interpret-version"
                  >
                    1.0.0
                  </span>
                </div>
              </div>
              <h3
                class="MuiTypography-root MuiTypography-titleLargeSemiBold emotion-19"
                data-testid="ai-interpret-example-name"
              >
                case name
              </h3>
              <div
                class="MuiStack-root emotion-20"
              >
                <div
                  class="MuiBox-root emotion-21"
                >
                  <span
                    class="MuiTypography-root MuiTypography-labelLargeSemiBold emotion-22"
                    data-testid="ai-interpret-gender-tag"
                  >
                    Female
                  </span>
                </div>
                <div
                  class="MuiBox-root emotion-21"
                >
                  <span
                    class="MuiTypography-root MuiTypography-labelLargeSemiBold emotion-22"
                    data-testid="ai-interpret-age-tag"
                  >
                    Age 29
                  </span>
                </div>
              </div>
            </div>
            <div
              class="MuiStack-root emotion-25"
            >
              <div
                class="emotion-26"
              />
              <div
                class="MuiStack-root emotion-27"
              >
                <svg />
                <span
                  class="MuiTypography-root MuiTypography-labelMedium emotion-28"
                  data-testid="ai-interpret-title"
                >
                  แปลและสรุปข้อมูลได้ดังนี้
                </span>
              </div>
            </div>
            <div
              class="MuiStack-root emotion-29"
            >
              <div
                class="MuiStack-root emotion-30"
              >
                <span
                  class="MuiTypography-root MuiTypography-labelExtraLargeSemiBold emotion-31"
                  data-testid="ai-interpret-title-0"
                >
                  1. title-ai-result
                </span>
                <p
                  class="MuiTypography-root MuiTypography-body1 emotion-32"
                  data-testid="ai-interpret-description-0"
                >
                  description-ai-result
                </p>
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium emotion-3"
                  data-testid="ai-interpret-button-copy-0"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiButton-startIcon MuiButton-iconSizeMedium emotion-4"
                  >
                    <svg />
                  </span>
                  Copy
                  <span
                    class="MuiTouchRipple-root emotion-5"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
